kafka/producer/record_accumulator.py

```diff
@@ -526,8 +526,12 @@ def await_flush_completion(self, timeout=None):
             for batch in self._incomplete.all():
                 log.debug('Waiting on produce to %s',
                           batch.produce_future.topic_partition)
-                assert batch.produce_future.wait(timeout=timeout), 'Timeout waiting for future'
-                assert batch.produce_future.is_done, 'Future not done?'
+                try:
+                    assert batch.produce_future.wait(timeout=timeout), 'Timeout waiting for future'
+                    assert batch.produce_future.is_done, 'Future not done?'
+                except AssertionError as e:
```

Comment:
might as well simply drop the asserts and use `if not`

though technically if wait() returns True, but is_done is False then it is not a Timeout but an unexpected error state, for which we could raise a timeout error but we also might pick some other more obviously bad exception.